src/Sulu/Bundle/CoreBundle/Listener/ControllerListener.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace PoolAlpin\Bundle\BaseBundle\Listener;
+
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\Common\Util\ClassUtils;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Sulu\Component\Validation\JsonSchema\Validate;
+use Sulu\Component\Validation\JsonSchema\Exceptions\SchemaValidationException;
+
+class ControllerListener
+{
+    private $annotationReader;
+
+    public function __construct(Reader $annotationReader)
+    {
+        $this->annotationReader = $annotationReader;
+    }
+
+    /**
+     * onKernelController
+     *
+     * Checks for annotations of type @Validate
+     *
+     * @param FilterControllerEvent $event
+     */
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $controller = $event->getController();
+
+        list($object, $method) = $controller;
+
+        // the controller could be a proxy, e.g. when using the JMSSecuriyExtraBundle or JMSDiExtraBundle
+        $className = ClassUtils::getClass($object);
+        $reflectionClass = new \ReflectionClass($className);
```

Comment:
I'm personally not a big fan of Reflection. Is there another possibility to solve this problem?